entity-service/src/test/kotlin/com/egm/stellio/entity/web/EntityOperationHandlerTests.kt

```diff
@@ -1,167 +1,231 @@
-
-        webClient.post()
-            .uri("/ngsi-ld/v1/entityOperations/create")
-            .header("Link", "<$aquacContext>; rel=http://www.w3.org/ns/json-ld#context; type=application/ld+json")
-            .bodyValue(jsonLdFile)
-            .exchange()
-            .expectStatus().isOk
-            .expectBody().json("{\n" +
-                    "    \"errors\": [\n" +
-                    "        {\n" +
-                    "            \"entityId\": \"urn:ngsi-ld:Sensor:HCMR-AQUABOX1dissolvedOxygen\",\n" +
-                    "            \"error\": [\n" +
-                    "                \"Entity already exists\"\n" +
-                    "            ]\n" +
-                    "        }\n" +
-                    "    ],\n" +
-                    "    \"success\": [\n" +
-                    "        \"urn:ngsi-ld:Sensor:HCMR-AQUABOX1temperature\",\n" +
-                    "        \"urn:ngsi-ld:Device:HCMR-AQUABOX1\"\n" +
-                    "    ]\n" +
-                    "}")
-    }
-
-    @Test
-    fun `create batch entity should return a 400 if JSON-LD payload is not correct`() {
-        val jsonLdFile = ClassPathResource("/ngsild/hcmr/HCMR_test_file_missing_context.json")
-
-        webClient.post()
-            .uri("/ngsi-ld/v1/entityOperations/create")
-            .header("Link", "<http://easyglobalmarket.com/contexts/diat.jsonld>; rel=http://www.w3.org/ns/json-ld#context; type=application/ld+json")
-            .bodyValue(jsonLdFile)
-            .exchange()
-            .expectStatus().isBadRequest
-            .expectBody().json("{\"type\":\"https://uri.etsi.org/ngsi-ld/errors/BadRequestData\"," +
-                    "\"title\":\"The request includes input data which does not meet the requirements of the operation\"," +
-                    "\"detail\":\"Could not parse entity due to invalid json-ld payload\"}")
-    }
-}
+package com.egm.stellio.entity.web
+
+import com.egm.stellio.entity.TestContainersConfiguration
+import com.egm.stellio.entity.service.EntityService
+import com.ninjasquad.springmockk.MockkBean
+import io.mockk.every
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.context.annotation.Import
+import org.springframework.core.io.ClassPathResource
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.reactive.server.WebTestClient
+
+@WithMockUser
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test")
+@AutoConfigureWebTestClient(timeout = "30000")
+class EntityOperationHandlerTests {
```

Comment:
The `EntityOperationService` should also be mocked here.

Reply:
I've commited a version in which I improved these tests.